### ISSUE TITLE
Compile Warnings with ecto_sql 3.2

### ISSUE THREAD

**Issue by jherdman** [gjaldon/ecto_enum#94]
I upgrade to ecto_sql 3.2 this morning and the following warnings are being raised when compiling my app:

```
warning: function embed_as/1 required by behaviour Ecto.Type is not implemented (in module AxisTypes)
  lib/figgis/enums.ex:3: AxisTypes (module)
```

**Comment by fschuindt** [comment]
I'm also getting this when compiling and using it:
```
==> ecto_enum
Compiling 5 files (.ex)
warning: function embed_as/1 required by behaviour Ecto.Type is not implemented (in module EctoEnum.Typespec.TestModule.StatusEnum)
  lib/ecto_enum/typespec.ex:26: EctoEnum.Typespec.TestModule.StatusEnum (module)

warning: function equal?/2 required by behaviour Ecto.Type is not implemented (in module EctoEnum.Typespec.TestModule.StatusEnum)
  lib/ecto_enum/typespec.ex:26: EctoEnum.Typespec.TestModule.StatusEnum (module)

warning: function embed_as/1 required by behaviour Ecto.Type is not implemented (in module EctoEnum.Typespec.TestModule.PGStatusEnum)
  lib/ecto_enum/typespec.ex:28: EctoEnum.Typespec.TestModule.PGStatusEnum (module)

warning: function equal?/2 required by behaviour Ecto.Type is not implemented (in module EctoEnum.Typespec.TestModule.PGStatusEnum)
  lib/ecto_enum/typespec.ex:28: EctoEnum.Typespec.TestModule.PGStatusEnum (module)

Generated ecto_enum app
==> database
Compiling 7 files (.ex)
warning: function embed_as/1 required by behaviour Ecto.Type is not implemented (in module GigsAPI.DataBase.Schemas.User.LevelEnum)
  lib/gigs_api/database/schemas/user/level_enum.ex:3: GigsAPI.DataBase.Schemas.User.LevelEnum (module)

warning: function equal?/2 required by behaviour Ecto.Type is not implemented (in module GigsAPI.DataBase.Schemas.User.LevelEnum)
  lib/gigs_api/database/schemas/user/level_enum.ex:3: GigsAPI.DataBase.Schemas.User.LevelEnum (module)

Generated database app
```

**Comment by jherdman** [comment]
This would be resolved by #92
